service/file.go

```diff
@@ -46,3 +48,34 @@ func FileCopy(src *model.Matter, parent string) error {
 	nm.Parent = parent
 	return gormutil.DB().Create(nm).Error
 }
+
+func CanMove(uid int64, parent string, file *model.Matter) (bool, error) {
+	var (
+		out []model.Matter
+	)
+	//check dir exists
+	if !MatterParentExist(uid, parent) {
+		return false, errors.New("dir does not exists")
+	}
+	//avoid move to itself
+	//eg: a/ -> a/
+	//eg: a/ -> a/b/c
+	//eg: b/ -> a/b/c
+	if parent != "" && strings.HasPrefix(parent, file.Parent+file.Name) {
+		return false, errors.New("can not move to itself")
+	}
+	//avoid move to same place
+	if parent == file.Parent {
+		return false, errors.New("file already in this dir")
+	}
+	//avoid dst dir has the same name file
```

Comment:
you can add the `file.name` into the query condition.

In fact, `service. MatterExist` has implemented

Reply:
yes,i notice `service. MatterExist`,but i think it's not necessary to access db again.